service/history/historyEngine.go

```diff
@@ -498,12 +498,13 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 				domainFailoverNotificationVersion := nextDomain.GetFailoverNotificationVersion()
 				domainActiveCluster := nextDomain.GetReplicationConfig().ActiveClusterName
 				previousFailoverVersion := nextDomain.GetPreviousFailoverVersion()
+				previousClusterName, _ := e.clusterMetadata.ClusterNameForFailoverVersion(previousFailoverVersion)
```

Comment:
I would not suppress the error here and log it at the very least. If I'm reading this correctly the callback will be messed up

Reply:
you're right